### PR TITLE
[NPUW]Fix VLM accuracy issue on  NPU.

### PR DESCRIPTION
**Pull request by intelgaoxiong**
### Details:
1. Not restore & store KV when it's a partial chunk. KV are calculated with padding data and will cause very slight results with float calculation.
2. VLM uses 3D position ID, position ID tensor has stride for chunk prefill inference. Changed to process 3D position ID correctly.

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp [hunk 1]
```diff
@@ -381,6 +385,12 @@ void ov::npuw::LLMInferRequest::prepare_for_new_conversation() {
     }
     uu::fill_tensor<int64_t>(m_prefill_request->get_tensor(m_prefill_in_ports.at(layer_names::attention_mask)), 0);
     uu::fill_tensor<int64_t>(m_prefill_request->get_tensor(m_prefill_in_ports.at(layer_names::position_ids)), 0);
+
+    // Clear all past_key_values tensors - use cached ports for efficiency
+    for (const auto& port : m_prefill_past_kv_ports) {
+        uu::fill_tensor_bytes(m_prefill_request->get_tensor(port), 0u);
+    }
```
Comment (dmatveev) [inline src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp:392]: This may impact TTFT, especially with the long prompts. That `fill_tensor_bytes` ins't very fast. Why do we need to have them all 0s? With the proper attention mask the unititialized values shouldn't impact the end result in the end?

File: src/plugins/intel_npu/src/plugin/npuw/llm_prefix_caching.cpp [hunk 2]
```diff
@@ -315,50 +315,6 @@ void PrefixCachingHelper::store_computed_blocks(size_t chunk_size,
...
-    // tokens. Round 3:
-    // - 256 + 256 tokens are stored in the past KV.
-    // - Infer for 256 tokens in the present KV.
-    // - After updating the KV cache, the past KV will hold 128 + 128 + 256 + 256 tokens, leaving 896 - 128 - 256 - 256
-    // = 256 tokens. Round 4:
-    // - 128 + 128 + 256 + 256 tokens are stored in the past KV.
-    // - Infer for the last 256 tokens in the present KV, completing the prefill for all prompts.
-
-    return chunk_len - restored_token_num % chunk_len;
-}
```
Comment (dmatveev) [inline src/plugins/intel_npu/src/plugin/npuw/llm_prefix_caching.cpp:360]: Where did all this complexity go?

File: src/plugins/intel_npu/src/plugin/npuw/llm_prefix_caching.cpp [hunk 3]
```diff
@@ -420,20 +376,23 @@ void PrefixCachingHelper::create_name_mapping() {
     }
 }
 
-uint64_t PrefixCachingHelper::restore_blocks(const ov::SoPtr<ov::ITensor>& input_ids,
-                                             const std::vector<uint64_t>& prompt_hashes) {
+std::vector<std::shared_ptr<KVBlock>> PrefixCachingHelper::find_cached_blocks(
+    const ov::SoPtr<ov::ITensor>& input_ids,
+    const std::vector<uint64_t>& prompt_hashes) {
     const uint64_t block_size = m_request.m_npuw_llm_compiled_model->m_prefix_caching_block_size;
+    const uint64_t chunk_size = m_request.m_npuw_llm_compiled_model->m_prefill_chunk_size;
```
Comment (dmatveev) [inline src/plugins/intel_npu/src/plugin/npuw/llm_prefix_caching.cpp:383]: What if chunking is not used? or is it mandatory to be used now with caching?

File: src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp (hunk 1 above)
Comment (intelgaoxiong) [reply src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp:392]: With the proper attention mask the uninitialized values may cause slight bitwise difference.
In most case, the detokenized output is not impacted.
The unitized values improve numerical stability.
Whether we can tolerate these bitwise differences is a matter for discussion.

File: src/plugins/intel_npu/src/plugin/npuw/llm_prefix_caching.cpp (hunk 2 above)
Comment (intelgaoxiong) [reply src/plugins/intel_npu/src/plugin/npuw/llm_prefix_caching.cpp:360]: adjust_chunk_size for prefix caching is not needed any more given we don't restore from cache for partial chunk now.

File: src/plugins/intel_npu/src/plugin/npuw/llm_prefix_caching.cpp (hunk 3 above)
Comment (intelgaoxiong) [reply src/plugins/intel_npu/src/plugin/npuw/llm_prefix_caching.cpp:383]: Prefix caching has to be working with chunk prefill so that we can manage those "past_key_value" nodes.
For the static prefill, there're no "past_key_value" nodes in IR.

File: src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp (hunk 1 above)
Comment (intelgaoxiong) [reply src/plugins/intel_npu/src/plugin/npuw/llm_infer_request.cpp:392]: We can also implement `fill_tensor_bytes` with `memset`, instead of `std::fill_n`
`memset` is much faster.
@dmatveev
